pkg/controllers/addons_controller.go

```diff
@@ -39,224 +36,120 @@ import (
 // AddonsLayerReconciler reconciles a AddonsLayer object.
 type AddonsLayerReconciler struct {
 	client.Client
-	Log     logr.Logger
-	Scheme  *runtime.Scheme
-	Applier apply.LayerApplier
+	Config   *rest.Config
+	k8client kubernetes.Interface
+	Log      logr.Logger
+	Scheme   *runtime.Scheme
+	Context  context.Context
+	Applier  apply.LayerApplier
 }
 
 // NewReconciler returns an AddonsLayerReconciler instance
-func NewReconciler(client client.Client, logger logr.Logger,
+func NewReconciler(config *rest.Config, client client.Client, logger logr.Logger,
 	scheme *runtime.Scheme) (reconciler *AddonsLayerReconciler, err error) {
 	reconciler = &AddonsLayerReconciler{
+		Config: config,
 		Client: client,
 		Log:    logger,
 		Scheme: scheme,
 	}
-	reconciler.Applier, err = apply.NewApplier(logger)
+	reconciler.k8client = reconciler.getK8sClient()
+	reconciler.Context = context.Background()
+	reconciler.Applier, err = apply.NewApplier(client, logger, scheme)
 	return reconciler, err
 }
 
-// GetK8sClient gets the Kubernetes client.
-func GetK8sClient() (*kubernetes.Clientset, error) {
-	kubeConfig := os.Getenv("KUBECONFIG")
-	if len(kubeConfig) > 0 {
-		// use the current context in kubeconfig
-		config, err := clientcmd.BuildConfigFromFlags("", kubeConfig)
-		if err != nil {
-			return nil, err
-		}
-
-		// create the clientset
-		clientset, err := kubernetes.NewForConfig(config)
-		if err != nil {
-			return nil, err
-		}
-
-		return clientset, nil
-	}
-	// creates the in-cluster config
-	config, err := rest.InClusterConfig()
-	if err != nil {
-		panic(err.Error())
-	}
+func (r *AddonsLayerReconciler) getK8sClient() kubernetes.Interface {
 	// creates the clientset
-	clientset, err := kubernetes.NewForConfig(config)
+	clientset, err := kubernetes.NewForConfig(r.Config)
 	if err != nil {
+		//TODO - Adjust error handling?
 		panic(err.Error())
 	}
 
-	return clientset, nil
-}
-
-func logBadStatusError(log logr.Logger, status string) {
-	utils.Log(log, 3, 1, fmt.Sprintf("invalid AddonLayer status: %s, ignoring", status))
+	return clientset
 }
 
-func processEmptyStatus(l *layers.Layer) {
-	utils.Log(l.GetLogger(), 2, 1, "processing", "Status", "")
-	l.StatusPrunePending()
-	l.SetRequeue()
-	utils.Log(l.GetLogger(), 2, 1, "processed",
-		"Previous Status", "",
-		"Status", l.GetFullStatus(),
-		"Spec:", l.GetSpec())
-}
+func (r *AddonsLayerReconciler) processAddonLayer(l layers.Layer) error {
+	utils.Log(r.Log, 1, 1, "processing", "Name", l.GetName(), "Status", l.GetStatus())
```

Comment:
Not sure you need wrapper functions for log, can you not just call r.Log directly? If you really want wrapper functions then they should be moved to a `logging` package.